src/client_builder.rs

```diff
@@ -61,49 +160,85 @@ where
         ClientBuilder {
             domain,
             port,
-            starttls: false,
+            #[cfg(any(feature = "native-tls", feature = "rustls-tls"))]
+            mode: ConnectionMode::AutoTls,
+            #[cfg(all(not(feature = "native-tls"), not(feature = "rustls-tls")))]
+            mode: ConnectionMode::Auto,
```

Comment:
One thing I worry about here is that if someone pulls in this dependency with `default-features = false`, they might not ever realize that they're doing plaintext connections. I wonder if the default mode if not TLS feature is enabled should be one that will _always_ error saying "No TLS implementation available". And we should require an _explicit_ `.mode(ConnectionMode::Plaintext)`.

Reply:
Which, I suppose, is really just saying that I think the default should _always_ be `AutoTls`.